lib/ret/login.ex

```diff
@@ -22,4 +24,22 @@ defmodule Ret.Login do
     |> unique_constraint(:identifier_hash)
     |> put_assoc(:account, account)
   end
+
+  def update_identifier_hash(%{old_email: old_email, new_email: new_email} = _) do
+    old_identifier_hash = Account.identifier_hash_for_email(old_email)
+    new_identifier_hash = Account.identifier_hash_for_email(new_email)
+
+    login_to_update = Repo.get_by(Login, identifier_hash: old_identifier_hash)
+
+    case login_to_update do
+      %Login{} ->
+        login_to_update
+        |> cast(%{identifier_hash: new_identifier_hash}, [:identifier_hash])
+        |> unique_constraint(:identifier_hash)
```

Comment:
Is the idea here that if an account already exists for new_email this will just fail? I suspect we will need to be able to handle this case upstream so we might not want to just return a 500.

Reply:
Yes, and that's a good point. I added some useful error messages to disambiguate failures:
- `:new_email_already_in_use`
- `:no_account_for_old_email`
- a catch-all for unknown error

https://github.com/mozilla/reticulum/pull/636/commits/0977f6635632cb2cdd1831fce6c74327f5adf847